### PR TITLE
fulltext: handle errors in pipeline

### PR DESCRIPTION
**Pull request by MJedr**

### CODE REVIEW

File: backend/tests/unit/records/marshmallow/literature/test_es_schema.py [hunk 1]
```diff
@@ -91,3 +91,39 @@ def test_es_schema_removes_fulltext_fields_from_ui_display(
...
+                "attachment": {"content": "this is a text"},
+                "error": {
+                    "message": "Fulltext indexing failed with message expected='>' actual='À' at offset 102005"
+                },
+            }
+        ]
+    }
+    serialized = schema.dump(entry_data).data
+    result = orjson.loads(schema.dumps(serialized).data)["_ui_display"]
+    assert '"error":' not in result
```
Comment (drjova) [inline backend/tests/unit/records/marshmallow/literature/test_es_schema.py:129]:
```suggestion
    assert "error" not in result
```

File: backend/inspirehep/records/marshmallow/literature/ui.py [hunk 2]
```diff
@@ -256,6 +256,8 @@ def get_documents_without_fulltext(self, data):
                 del document["attachment"]
             if "text" in document:
                 del document["text"]
+            if "error" in document:
+                del document["error"]
```
Comment (drjova) [inline backend/inspirehep/records/marshmallow/literature/ui.py:260]: I would put it in the Base literature marshamallow to remove it from all API responses.

Comment (MJedr) [reply backend/inspirehep/records/marshmallow/literature/ui.py:260]: so maybe let's use private field there, these fields are hidden in public schema then 
https://github.com/inspirehep/inspirehep/blob/243f4378683d4ba6357b3adec76d911f8f1422b3/backend/inspirehep/records/marshmallow/literature/base.py#L51

Comment (drjova) [reply backend/inspirehep/records/marshmallow/literature/ui.py:260]: I think it does something else https://github.com/inspirehep/inspirehep/blob/243f4378683d4ba6357b3adec76d911f8f1422b3/backend/inspirehep/records/marshmallow/fields/non_hidden.py#L11-L15

**Review by drjova — COMMENTED** [review]
LGTM, shouldn't we update the mappings as well though?

**Comment by MJedr** [comment]
yes, good point 👍

**Review by drjova — APPROVED** [review]

**Pull request merged**